app/pkg/build.gradle

```diff
@@ -0,0 +1,110 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+plugins {
+    id 'java'
+    id "de.jjohannes.extra-java-module-info" version "0.11"
+    id "org.beryx.jlink" version "2.24.1"
+}
+
+repositories {
+    mavenCentral()
+}
+
+dependencies {
+    // Note that we depend on the :app:gui shadow jar, which bundles the gui
+    // application and all its dependencies into a single uber jar.
+    implementation project(path: ':app:gui', configuration: 'shadow')
+}
+
+extraJavaModuleInfo {
+    // JPackage (below) requires that we package our application as a Java
+    // Module.  Instead of attempting to modularize the entire gtfs-validator
+    // project, we instead make this project a module and moduarlize our single
+    // :app:gui uber-jar dependency by injecting a module-info.class entry into
+    // the jar with its native dependencies.
+    //
+    // See additional discussion in https://bit.ly/gtfs-validator-packaged-exe and
+    // https://docs.gradle.org/current/samples/sample_java_modules_with_transform.html
+    module('gui-all.jar', 'org.mobilitydata.gtfsvalidator.app.gui', project.version) {
+        exports('org.mobilitydata.gtfsvalidator.app.gui')
+
+        // This is the set of core Java modules that our application depends on.
+        // This list was hand-curated by looking at the output of `jdeps -s` on
+        // the full application jar, which produces some false positives given
+        // that not all code-paths in our dependencies are actually used.  I
+        // have also run the app with no module dependencies to see what kind
+        // of exceptions we get.
+        requires('java.compiler')
+        requires('java.desktop')
+        requires('java.logging')
+        requires('java.naming')
+        requires('java.security.jgss')
+        requires('java.sql')
+  }
+}
+
+sourceCompatibility = JavaVersion.VERSION_11
+compileJava.options.encoding = "UTF-8"
+
+java {
+    modularity.inferModulePath = true
+}
+
+application {
+    mainClass = 'org.mobilitydata.gtfsvalidator.app.pkg.Main'
+    mainModule = 'org.mobilitydata.gtfsvalidator.app.pkg'
+}
+
+jar {
+    // Add the manifest within the JAR, using gtfs-validator as the title
+    manifest {
+        attributes('Implementation-Title': rootProject.name,
+                'Implementation-Version': project.version)
+    }
+}
+
+// Debugging tips:
+// If you ever get an error when running the `jpackage` task like:
+//   Error reading module: app/pkg/build/jlinkbase/jlinkjars/pkg.jar
+// (and not much else), rerun the Gradle task with --debug specified and look
+// for the full command-line executed for `jlink` or `jpackage`.  Copy the
+// command-line and run it directly with a `-J-Djlink.debug=true` flag added
+// and you will get more useful information about what went wrong.
+jlink {
+    moduleName = 'org.mobilitydata.gtfsvalidator.app.pkg'
+    launcher {
+        name = 'GTFS Validator'
```

Comment:
I built the exe and it worked!

![image](https://user-images.githubusercontent.com/928045/167673115-0eae617b-5f4c-400d-8354-306648b2f954.png)

One item - the version reflected here is `1.0`, which doesn't match the actual project version. However, this is a little tricky as the version number is defined by the Git tag in the release process (see RELEASE.md). 

In `main/build.gradle`, it looks like this is passed via an environmental variable - `version = "${System.getenv("versionTag")}"`. Could we use the same process here, and just pass the the version in manually for now when building package from command line (and update the BUILD.md docs accordingly)? 

I believe we could automate the build of the packaged executable with a GitHub Action as well (including grabbing https://github.com/wixtoolset/wix3/releases/tag/wix3112rtm), although I'd suggest we do that in a separate PR.

Reply:
This is unfortunately more complicated :)

One immediate issue is with the formatting of version tags in `gtfs-validator`.  Namely, I want to call out that the use of the 'v' prefix is going to be problematic, because many tools (e.g. Java Modules, Mac OS Package Files) expect a purely numeric version number.

I'm also wondering if we can do better than the versionTag env variable.  It seems non-idomatic from a Gradle stanpoint.  It seems like overriding the project.version via a command-line property (-Pversion=1.2.3) seems like a more standard way of doing things?  Indeed, the jpackage command correctly picks up and uses project.version if it's correctly specified (and defaults to 1.0 if not).

Thoughts?

Reply:
>Namely, I want to call out that the use of the 'v' prefix is going to be problematic, because many tools (e.g. Java Modules, Mac OS Package Files) expect a purely numeric version number.

If this is getting in the way of the other tooling, then 👍  from me for changing it.

>It seems like overriding the project.version via a command-line property (-Pversion=1.2.3) seems like a more standard way of doing things?

👍 , especially if jpackage automatically picks this up. From a quick scan it looks like this would need to change across many of the GitHub Actions .yml files - https://github.com/MobilityData/gtfs-validator/search?q=VERSION_TAG%3Dv

Given that both above changes are more invasive, how about saving them to new PRs so we can get this merged?

Reply:
> Given that both above changes are more invasive, how about saving them to new PRs so we can get this merged?

I agree. I'm okay with having a`1.0` version for the exe/msi that doesn't match the validator version for now, and then we'll fix that in another PR.

> Namely, I want to call out that the use of the 'v' prefix is going to be problematic, because many tools (e.g. Java Modules, Mac OS Package Files) expect a purely numeric version number.

I agree, we can change it.

Reply:
I captured this discussion in issue #1139.  I'm going to resolve this for now.